File: backend/src/FriendZonePlus.Core/Entities/WallPost.cs
namespace FriendZonePlus.Core.Entities;

public class WallPost
{
  public int Id { get; set; }
  public string Content { get; set; } = string.Empty;
  public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

  //Relationships
  //User who authored the post
  public int AuthorId { get; set; }
  public User Author { get; set; } = null!;
  //Target user for post
  public int TargetUserId { get; set; }
  public User TargetUser { get; set; } = null!;
}

File: frontend/types/wallpost.ts
export interface CreateWallPostDto {
    targetUserId: number;
    content: string;
}

export interface UpdateWallPostDto {
    content: string;
}

export interface WallPostResponseDto {
    id: number;
    content: string;
    createdAt: string;
    authorId: number;
    authorName: string;
    targetUserId: number;
    targetUserName: string;
}